src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvokerProvider.cs

```diff
@@ -32,6 +34,7 @@ public class ControllerActionInvokerProvider : IActionInvokerProvider
             _maxModelValidationErrors = optionsAccessor.Value.MaxModelValidationErrors;
             _logger = loggerFactory.CreateLogger<ControllerActionInvoker>();
             _diagnosticSource = diagnosticSource;
+            _combineAuthorizeFilters = optionsAccessor.Value.CombineAuthorizeFilters;
```

Comment:
Instead of plumbing this value through, could we use an ApplicationModelProvider and have it produce an effective `AuthorizationFilter` that does this? Am I missing some a scenario where this approach wouldn't work?